src/Microsoft.Net.Http.Headers/HeaderUtilities.cs

```diff
@@ -594,13 +595,81 @@ public static string FormatDate(DateTimeOffset dateTime, bool quoted)
 
         public static StringSegment RemoveQuotes(StringSegment input)
         {
-            if (!StringSegment.IsNullOrEmpty(input) && input.Length >= 2 && input[0] == '"' && input[input.Length - 1] == '"')
+            if (IsQuoted(input))
             {
                 input = input.Subsegment(1, input.Length - 2);
             }
             return input;
         }
 
+        public static bool IsQuoted(StringSegment input)
+        {
+            return !StringSegment.IsNullOrEmpty(input) && input.Length >= 2 && input[0] == '"' && input[input.Length - 1] == '"';
+        }
+
+        public static StringSegment DecodeEscapeCharacters(StringSegment input)
```

Comment:
Yeah I was thinking we needed a rename considering "Encode" and "Decode" are too broad.